File: modules/spring-core/testdata/inspection/methodsAliasFor/src/com/MethodAliasForConfiguration.java
package com

import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.core.annotation.AliasFor;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

@Configuration
public class MethodAliasForConfiguration {

    @InheritorAnnotation
    @Bean
    MethodAliasClass someMethod() {}
    
}

class MethodAliasClass {}

@Target({ElementType.METHOD, ElementType.ANNOTATION_TYPE})
@Retention(RetentionPolicy.RUNTIME)
@interface AncestorAnnotation {
    @AliasFor("value")
    String[] name() default {};
    
    @AliasFor("name")
    String[] value() default {};
    
}

@Target({ElementType.METHOD, ElementType.ANNOTATION_TYPE})
@Retention(RetentionPolicy.RUNTIME)
@interface OtherAnnotation {
    @AliasFor("value")
    String[] name() default {};
    
    @AliasFor("name")
    String[] value() default {};
}

@Target({ElementType.METHOD, ElementType.ANNOTATION_TYPE})
@Retention(RetentionPolicy.RUNTIME)
@AncestorAnnotation
@interface InheritorAnnotation {
    
    private final static java.lang.String EXISTING_METHOD = "value";
    private final static java.lang.String NOT_EXISTING_METHOD = "notExistingMethod";
    
    @AliasFor("unknown")
    String[] name() default {};
    
    @AliasFor(annotation=AncestorAnnotation.class, attribute="value")
    String[] value() default {};
    
    @AliasFor(annotation=AncestorAnnotation.class, attribute=EXISTING_METHOD)
    String[] value2() default {};
    
    @AliasFor(annotation=AncestorAnnotation.class, attribute=NOT_EXISTING_METHOD)
    String[] unknownConstForAncestor() default {};
    
    @AliasFor(annotation=AncestorAnnotation.class, attribute="unknownForAncestor")
    String[] unknownForAncestor() default {};
    
    @AliasFor(annotation=OtherAnnotation.class, attribute="value")
    String[] notMetaAnnotated() default {};
    
}

File: modules/spring-core/testdata/kotlin/inspection/methodsAliasFor/src/com/MethodAliasForConfiguration.kt
package com

import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.core.annotation.AliasFor

@Configuration
open class MethodAliasForConfiguration {
    @InheritorAnnotation
    @Bean
    open fun someMethod(): MethodAliasClass? {
        return null
    }
}

class MethodAliasClass

@Target(
    AnnotationTarget.FUNCTION,
    AnnotationTarget.PROPERTY_GETTER,
    AnnotationTarget.PROPERTY_SETTER,
    AnnotationTarget.ANNOTATION_CLASS
)
@Retention(
    AnnotationRetention.RUNTIME
)
internal annotation class AncestorAnnotation(
    @get:AliasFor("value") val name: Array<String> = [],
    @get:AliasFor("name") vararg val value: String = []
)

@Target(
    AnnotationTarget.FUNCTION,
    AnnotationTarget.PROPERTY_GETTER,
    AnnotationTarget.PROPERTY_SETTER,
    AnnotationTarget.ANNOTATION_CLASS
)
@Retention(
    AnnotationRetention.RUNTIME
)
internal annotation class OtherAnnotation(
    @get:AliasFor("value") val name: Array<String> = [],
    @get:AliasFor("name") vararg val value: String = []
)

@Target(
    AnnotationTarget.FUNCTION,
    AnnotationTarget.PROPERTY_GETTER,
    AnnotationTarget.PROPERTY_SETTER,
    AnnotationTarget.ANNOTATION_CLASS
)
@Retention(
    AnnotationRetention.RUNTIME
)
@AncestorAnnotation
internal annotation class InheritorAnnotation(
    @get:AliasFor("unknown") val name: Array<String> = [],
    @get:AliasFor(
        annotation = AncestorAnnotation::class,
        attribute = "value"
    ) vararg val value: String = [],
    @get:AliasFor(
        annotation = AncestorAnnotation::class,
        attribute = EXISTING_METHOD
    ) val value2: Array<String> = [],
    @get:AliasFor(
        annotation = AncestorAnnotation::class,
        attribute = NOT_EXISTING_METHOD
    ) val unknownConstForAncestor: Array<String> = [],
    @get:AliasFor(
        annotation = AncestorAnnotation::class,
        attribute = "unknownForAncestor"
    ) val unknownForAncestor: Array<String> = [],
    @get:AliasFor(
        annotation = OtherAnnotation::class,
        attribute = "value"
    ) val notMetaAnnotated: Array<String> = []
) {
    companion object {
        const val EXISTING_METHOD: String = "value"
        const val NOT_EXISTING_METHOD: String = "notExistingMethod"
    }
}